### PR TITLE
add Report

### PR DESCRIPTION
**Pull request by jessejanderson**

### CODE REVIEW

File: lib/pco_api/people/report.ex [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+  use PcoApi.Actions
+
+  def list(params) when is_list(params), do: get(params, "reports")
+
+  def get(id) when is_integer(id), do: get("reports/#{id}")
+
+  def create(%PcoApi.Record{attributes: _, type: "Report"} = record), do: create(record, "reports")
+
+  def self(%PcoApi.Record{id: id}), do: get "reports/#{id}"
```
Comment (geolessel) [inline lib/pco_api/people/report.ex]: I know there are few examples _not_ like what I'm about to suggest in the codebase, but let's pattern match on the type of record:

``` elixir
def self(%PcoApi.Record{type: "Report", id: id}), do: get "reports/#{id}"
```

File: test/pco_api/people/report_test.exs [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    Bypass.expect bypass, fn conn ->
+      assert "/people/v2/people/1" == conn.request_path
+      Plug.Conn.resp conn, 200, Fixture.read("report.json")
+    end
+    assert %PcoApi.Record{} = bypass |> record_with_links |> Report.created_by
+  end
+
+  test ".updated_by gets list updated_by", %{bypass: bypass} do
+    Bypass.expect bypass, fn conn ->
+      assert "/people/v2/people/1" == conn.request_path
```
Comment (geolessel) [inline test/pco_api/people/report_test.exs]: Can you make this a different ID from `created_by` just to ensure the functions are getting their info from different links?

File: test/pco_api/people/report_test.exs [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    %PcoApi.Record{
+      type: "Report",
+      links: %{
+        "created_by" => "http://localhost:#{bypass.port}/people/v2/people/1",
+        "updated_by" => "http://localhost:#{bypass.port}/people/v2/people/1",
+        "self" => "http://localhost:#{bypass.port}/people/v2/reports/1"
+      }
+    }
+  end
+end
```
Comment (geolessel) [inline test/pco_api/people/report_test.exs:99]: Mind adding a `.new` test?

File: lib/pco_api/people/report.ex (hunk 1 above)
Comment (jessejanderson) [inline lib/pco_api/people/report.ex]: 👍

**Comment by geolessel** [comment]
👍

**Pull request merged**